### PR TITLE
Add notification to curators when an item is marked complete

### LINKED ISSUE #505: Error while editing an approved work
I came across this error when adding some tests for the edit of  an approved work.  I will be putting a pending test into a PR soon...

[pdc_describe/staging] NoMethodError: undefined method `id' for #<S3File:0x00007fe4e8789a18 @filename="10.80021/sxes-6k67/84/2022-09-05_09-31-44_834.heic", @last_modified=2022-09-29 18:28:26 UTC, @size=2746794, @checksum="59c4a9c70640148e7c8273140f5e6bac", @query_service=#<S3QueryService:0x00007fe4e8740048 @model=#<Work id: 84, work_type: "DATASET", state: "approved", collection_id: 1, created_by_user_id: 14, created_at: "2022-09-09 17:13:10.759490000 +0000", updated_at: "2022-09-29 18:28:25.950793000 +0...

## Backtrace

    line 243 of [PROJECT_ROOT]/app/models/work.rb: block in uploads_attributes
    line 241 of [PROJECT_ROOT]/app/models/work.rb: map
    line 241 of [PROJECT_ROOT]/app/models/work.rb: uploads_attributes

## Root Cause
The root cause of this issue is that we are showing uploads [here](https://github.com/pulibrary/pdc_describe/blob/main/app/views/works/_form.html.erb#L44-L49).  Not pre-curation or post-curation, but just uploads.  Post-curation files are different from pre-curation files now and can (and should not) be displayed in the same way. 

We could easily solve the bug by not displaying the files on the edit form for approved works.  What should we really do though?  I feel like displaying the files (non editable) for curators, and maybe linking to Globus if they want to modify them seems like a reasonable first step.

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/99214/faults/89273390)

### PR DESCRIPTION
**Pull request by carolyncole**
fixes #505

### CODE REVIEW

File: app/models/work.rb [hunk 1]
```diff
@@ -723,5 +723,11 @@ def publish_precurated_files
 
       delete_pre_curation_s3_dir
     end
+
+    def notify_collection_curators(current_user)
+      curators = collection.administrators.map { |admin| "@#{admin.uid}" }.join(", ")
+      notification = "#{curators} The [work](#{work_path(self)}) is ready for review."
```
Comment (mccalluc) [inline app/models/work.rb:729]: I'm curious whether this is a full url or just a path... but that's something for me to track down.

File: spec/models/work_spec.rb [hunk 2]
```diff
@@ -548,6 +548,15 @@
...
       expect { awaiting_approval_work.draft!(user) }.to raise_error AASM::InvalidTransition
     end
+
+    it "notifies the curators it is ready for review" do
+      curator = FactoryBot.create(:research_data_moderator)
+      expect { awaiting_approval_work }.to change { WorkActivity.where(activity_type: "SYSTEM").count }.by(2)
+      expect(WorkActivity.where(activity_type: "SYSTEM").last.message).to eq("marked as awaiting_approval")
+      curator_notification = WorkActivity.where(activity_type: "SYSTEM").first.message
+      expect(curator_notification).to include("@#{curator.uid}")
+      expect(curator_notification). to include(Rails.application.routes.url_helpers.work_path(awaiting_approval_work))
```
Comment (mccalluc) [inline spec/models/work_spec.rb:558]:
```suggestion
      expect(curator_notification).to include(Rails.application.routes.url_helpers.work_path(awaiting_approval_work))
```

**Review by mccalluc — APPROVED** [review]

**Review by mccalluc — APPROVED** [review]

File: app/models/work.rb (hunk 1 above)
Comment (carolyncole) [reply app/models/work.rb:729]: Just a relative path. I see your point, a path might not be helpful in an email.

File: app/models/work.rb [hunk 3]
```diff
@@ -728,5 +728,11 @@ def publish_precurated_files
 
       delete_pre_curation_s3_dir
     end
+
+    def notify_collection_curators(current_user)
+      curators = collection.administrators.map { |admin| "@#{admin.uid}" }.join(", ")
+      notification = "#{curators} The [work](#{work_url(self)}) is ready for review."
```
Comment (carolyncole) [inline app/models/work.rb:734]: @mccalluc change this to the url so it will be more helpful in an email.

**Review by mccalluc — APPROVED** [review]

**Pull request merged**